crates/ide/src/runnables.rs

```diff
@@ -306,9 +313,10 @@ pub(crate) fn runnable_fn(sema: &Semantics<RootDatabase>, def: hir::Function) ->
     let name_string = def.name(sema.db).to_string();
 
     let root = def.module(sema.db).krate().root_module(sema.db);
-
     let kind = if name_string == "main" && def.module(sema.db) == root {
         RunnableKind::Bin
+    } else if name_string == "_start" && def.module(sema.db) == root {
+        RunnableKind::WasmBin
```

Comment:
Is there a more reliable way to detect this? Just checking for `_start` seems fragile

Reply:
In case of wasi you normally don't write `_start` either. You write `main` and then wasi-libc provides `_start` just like libc on any other architecture.